.editorconfig

```diff
@@ -267,6 +267,49 @@ dotnet_diagnostic.SA1026.severity = warning
 dotnet_diagnostic.SA1027.severity = none # RSPEC-105
 dotnet_diagnostic.SA1028.severity = warning
 
+# Readability Rules (SA1100-) https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/1.1.118/DOCUMENTATION.md
+
+dotnet_diagnostic.SA1100.severity = warning
+dotnet_diagnostic.SA1101.severity = none    # Doesn't match our coding style, we don't use "this." when not neded
+dotnet_diagnostic.SA1102.severity = warning
+dotnet_diagnostic.SA1103.severity = warning
+dotnet_diagnostic.SA1104.severity = warning
+dotnet_diagnostic.SA1105.severity = warning
+dotnet_diagnostic.SA1106.severity = warning
+dotnet_diagnostic.SA1107.severity = warning
+dotnet_diagnostic.SA1108.severity = none    # Noisy for short comments on short lines (if, for, foreach)
+dotnet_diagnostic.SA1109.severity = warning
+dotnet_diagnostic.SA1110.severity = warning
+dotnet_diagnostic.SA1111.severity = warning
+dotnet_diagnostic.SA1112.severity = warning
+dotnet_diagnostic.SA1113.severity = warning
+dotnet_diagnostic.SA1114.severity = none    # Prevents putting comment before first member in ImmutableArray.Create
+dotnet_diagnostic.SA1115.severity = warning
+dotnet_diagnostic.SA1116.severity = none    # Waste of new lines in simple scenarios
+dotnet_diagnostic.SA1117.severity = none    # Waste of new lines in simple scenarios
+dotnet_diagnostic.SA1118.severity = none    # Noisy in UTs
+dotnet_diagnostic.SA1119.severity = warning
+dotnet_diagnostic.SA1120.severity = warning
+dotnet_diagnostic.SA1121.severity = warning
+dotnet_diagnostic.SA1122.severity = warning
+dotnet_diagnostic.SA1123.severity = warning
+dotnet_diagnostic.SA1124.severity = none    # We need regions sometimes
+dotnet_diagnostic.SA1125.severity = warning
+dotnet_diagnostic.SA1126.severity = none    # Deprecated / not implemented rule
+dotnet_diagnostic.SA1127.severity = none    # Noisy for single-line method declarations
+dotnet_diagnostic.SA1128.severity = none    # Doesn't match our code base
+dotnet_diagnostic.SA1129.severity = warning
+dotnet_diagnostic.SA1130.severity = warning
+dotnet_diagnostic.SA1131.severity = warning
```

Comment:
I am curious on what this will find in our current codebase - we should also adapt our coding style to reflect this, right?

Reply:
Quite a few actually, as we had C++ and Java devs contributing and something got copy-paste spreaded :)
https://github.com/SonarSource/sonar-dotnet/blob/36b38ad5a4f08f580e3ccd90e0b1d960d3d2c701/analyzers/src/SonarAnalyzer.CSharp/CBDE/MlirExporter.cs#L289

https://github.com/SonarSource/sonar-dotnet/blob/36b38ad5a4f08f580e3ccd90e0b1d960d3d2c701/analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/CookieShouldBeSecure.cs#L45

Reply:
before enabling this one, shouldn't we discuss it and add it to our coding style? 

I agree to enable it and have a consistent way of doing it, but I am thinking that I myself might have added sometimes constants on the left, so I'd favor a discussion before to make sure we're all aligned before this pops up on PRs

 (so I suggest to disable it in this PR until we discuss)

Reply:
This one is quite obvious/no brainer even without discussion, as we're doing it like that anyway. The rule will just help with that so I'm for merging it as is.

Reply:
if you're confident we're aligned on this, please update the coding style as well

Reply:
I'm not sure "coding style" is "obvious"